pkg/function-manager/controller.go

```diff
@@ -27,6 +27,67 @@ type ControllerConfig struct {
 	ResyncPeriod time.Duration
 }
 
+type sourceEntityHandler struct {
```

Comment:
Do we need a separate control loop for the source?  We aren't resolving any state.  I could certainly see an interface to source, but I think the "control loop" may not be the right one.

Reply:
Right now the control loop is used to resolved orphaned source entities, i.e. when a source entity is added to the store but the function entity fails to be added, the source status is set to `DELETING` and updated which the controller will pick up and delete. 

The control loop could be eliminated if a hard delete is used instead when a source becomes orphaned.